src/components/chart/chart.tsx

```diff
@@ -422,4 +433,38 @@ export class Chart {
     private recalculateRangeData() {
         this.range = this.calculateRange();
     }
+
+    private handleClick = (event: MouseEvent) => {
+        const target = event.currentTarget as HTMLElement;
+        const index = target.dataset.index;
+        if (index === undefined) {
+            return;
+        }
+
+        const item = this.items[Number(index)];
+        if (!item.clickable) {
+            return;
+        }
+
+        event.stopPropagation();
+        this.itemSelected.emit(item);
+    };
```

Comment:
Would it work to define this method as 
```suggestion
    private handleClick = (item: any) => () => {
        if (!item.clickable) {
            return;
        }

        event.stopPropagation();
        this.itemSelected.emit(item);
    };
```

Then you could inject the item straight away in line 235 with `onClick={this.handleClick(item)}`. You could do the same for the keyDown event, and you don't need to search the item based on the index anymore.